### ISSUE TITLE
CSV rows with explicit intervals are incorrectly visualized as moving points

### ISSUE THREAD

**Issue by meh9** [TerriaJS/terriajs#2447]
There appears to be some problems with CSV timeseries data. I'm finding it hard to exactly put my finger on what is wrong, but I'll try my best.

Using the attached CSV file, I have 4 points with 3 different time intervals - Jan 1 to March 1, April 1 to June 1, and July 1 to Aug 1. The lat/lon varies for the intervals so we should see things moving.

Dragging it onto a Terria instance the behaviour I'm seeing is:

Jan 1 to March 1
The points come up where expected. 
However, as we move the timeslider through the interval they move halfway to their positions in the next interval, and at the end of the interval, i.e. March 2, they pop into their starting positions for April 1 and then continue to be displayed throughout the rest of March. The gradual colour change due to the new value does something similar.

The other time intervals do similar things.

The most unexpected and clearly wrong thing here is that the points pop into the positions for the next interval and continue to display *between* the time intervals - nothing should be displayed between intervals.

The movement and colour changes could be argued about what the correct behaviour is.

[timeseries_test.csv.txt](https://github.com/TerriaJS/terriajs/files/871663/timeseries_test.csv.txt)

**Comment by RacingTadpole** [comment]
Sorry to take so long to look at this. 

This seems to be a terria clock thing.  When I click on one of the gap dates (eg. between 1 Mar and 1 Apr) and look at the "currentTime" on the terria clock (eg. by putting a breakpoint in FeatureInfoSection.jsx's render and clicking a feature to bring up the feature info panel, then typing `this.props.catalogItem.terria.clock.currentTime`), it shows 1 Apr (Julian day 2457479), regardless of the date shown on the time slider.

@kring or @AlexGilleran - do you know if this is the intended behaviour of the terria clock for some reason?

**Comment by RacingTadpole** [comment]
So this could be a bigger problem, that #1328 has broken...
@meh9 - have you noticed any problems with this in other situations? eg. the solar direct exposure mentioned in the comments to #896 ?
Also for ref, in #946 we made the timeline snap to the nearest datapoint, which is why you never see the points disappear completely here.

**Comment by kring** [comment]
Here's a simplified snippet (just one point) of Mats's CSV:

```
start_date,end_date,id,lat,lon,value
2016-01-01T00:00:00Z,2016-03-01T00:00:00Z,a,-30,130,1
2016-04-01T00:00:00Z,2016-06-01T00:00:00Z,a,-30,131,2
2016-07-01T00:00:00Z,2016-08-01T00:00:00Z,a,-30,132,3
```

I would argue that the desired behavior here is no movement at all. The point should be displayed at exactly (130,-30) between 2016-01-01 and 2016-03-01, then it should disappear until 2016-04-01, then snap to (131,-30) between 2016-04-01 and 2016-06-01, etc.  When you're specifying a position over an interval, it seems nonsensical for it to move over that interval.  Do you agree @meh9 and @RacingTadpole?

**Comment by meh9** [comment]
@kring Yes, I agree that they should be static during the interval and snap to the new position.
